src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
+    /// A <see cref="Utf8JsonStreamReaderConverter{T}"/> to allow read JSON into <see cref="LockFileTargetLibrary"/>
+    /// </summary>
+    internal class Utf8JsonStreamLockFileTargetLibraryConverter : IUtf8JsonStreamReaderConverter<LockFileTargetLibrary>
+    {
+        private static readonly byte[] TypePropertyName = Encoding.UTF8.GetBytes("type");
+        private static readonly byte[] FrameworkPropertyName = Encoding.UTF8.GetBytes("framework");
+        private static readonly byte[] DependenciesPropertyName = Encoding.UTF8.GetBytes("dependencies");
+        private static readonly byte[] FrameworkAssembliesPropertyName = Encoding.UTF8.GetBytes("frameworkAssemblies");
+        private static readonly byte[] RuntimePropertyName = Encoding.UTF8.GetBytes("runtime");
+        private static readonly byte[] CompilePropertyName = Encoding.UTF8.GetBytes("compile");
+        private static readonly byte[] ResourcePropertyName = Encoding.UTF8.GetBytes("resource");
+        private static readonly byte[] NativePropertyName = Encoding.UTF8.GetBytes("native");
+        private static readonly byte[] BuildPropertyName = Encoding.UTF8.GetBytes("build");
+        private static readonly byte[] BuildMultiTargetingPropertyName = Encoding.UTF8.GetBytes("buildMultiTargeting");
+        private static readonly byte[] ContentFilesPropertyName = Encoding.UTF8.GetBytes("contentFiles");
+        private static readonly byte[] RuntimeTargetsPropertyName = Encoding.UTF8.GetBytes("runtimeTargets");
+        private static readonly byte[] ToolsPropertyName = Encoding.UTF8.GetBytes("tools");
+        private static readonly byte[] EmbedPropertyName = Encoding.UTF8.GetBytes("embed");
+        private static readonly byte[] FrameworkReferencesPropertyName = Encoding.UTF8.GetBytes("frameworkReferences");
+
+        private IList<PackageDependency> ReadPackageDependencyList(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                return new List<PackageDependency>(0);
```

Comment:
```suggestion
                return new Array.Empty<PackageDependency>();
```

No allocation for that. 

Applies to any other places you may have this.